federation/pkg/kubefed/util/util.go

```diff
@@ -285,27 +288,47 @@ func getRBACVersion(discoveryclient discovery.CachedDiscoveryInterface) (*schema
 		return nil, fmt.Errorf("Couldn't get clientset to create RBAC roles in the host cluster: %v", err)
 	}
 
+	// These are the RBAC versions we can speak
+	knownVersions := map[schema.GroupVersion]bool{
```

Comment:
So at other places, we assume that the server supports a specific version (like extensions/v1beta1.Deployments?). Why not do the same here?

Whenever we will add a new version, we will likely miss adding it here anyways.